src/libs/elektra/plugin.c

```diff
@@ -167,7 +167,7 @@ size_t elektraPluginGetFunction (Plugin * plugin, const char * name)
 	plugin->kdbGet (plugin, exports, pk);
 	ksRewind (exports);
 	keyAddBaseName (pk, "exports");
-	keyAddBaseName (pk, name);
+	keyAddName (pk, name); // we may not want to escape for hooks exports
```

Comment:
Hmm I don't think just blocking names starting with `.` is enough. I could still pass `blahblah/../../..`. I think we should check whether the sequence `..` is contained in the finished key name.

Reply:
What is the problem with the escaping keyAddBaseName does? What is the benefit of the whole change?

"we **may** not" does not sound very convincing

Reply:
Previously, this method only worked for names that were of the format:
```
system:/elektra/modules/<module name>/exports/<function name>
```

However, hooks exports have a new format:
```
system:/elektra/modules/<module name>/exports/hooks/<hook name>/<function name>
```

So if we want to load a hook function, we have to pass `hooks/<hook name>/<function name>` as an argument to this function. If we use `keyAddBaseName`, the composed key will be 
`
system:/elektra/modules/<module name>/exports/hooks\/<hook name>\/<function name>
`
which is obviously not the same.

Reply:
> I think we should check whether the sequence `..` is contained in the finished key name.

Actually, it would have to be `/../` anywhere, `../` at the start or `/..` at the end, because things like `..a` and `a..` are fine in theory. One way to check that would be to make a copy of `pk` before `keyAddName` and then to a `keyIsBelow` check afterwards.

However, I think we could just say `name` is not allowed to contain the character `.` at all. Since we are talking about function names here, IMO that limitation is fine.

Reply:
I'd still allow `.`, as we're not talking purely about function names in here, and `.` could be useful as a seperator.